app/src/main/java/com/willowtree/vocable/settings/AddPhraseKeyboardFragment.kt

```diff
@@ -0,0 +1,84 @@
+package com.willowtree.vocable.settings
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.core.view.isVisible
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.willowtree.vocable.BaseViewModelFactory
+import com.willowtree.vocable.R
+import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.room.Phrase
+import com.willowtree.vocable.settings.EditPhrasesKeyboardFragmentArgs.Companion.fromBundle
+import java.util.*
+
+class AddPhraseKeyboardFragment : EditKeyboardFragment() {
+
+    companion object {
+        private const val KEY_PHRASE = "KEY_PHRASE"
+
+        fun newInstance(phrase: Phrase?): EditPhrasesKeyboardFragment {
+            return EditPhrasesKeyboardFragment().apply {
+                arguments = Bundle().apply {
+                    putParcelable(KEY_PHRASE, phrase)
+                }
+            }
+        }
+    }
+
+    private lateinit var viewModel: AddPhraseViewModel
+    private val args: AddPhraseKeyboardFragmentArgs by navArgs()
+    private lateinit var category: Category
+    private var savedPhrase = ""
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        category = args.category
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.backButton.action = {
+            if (isDefaultTextVisible() || binding.keyboardInput.text.toString() == savedPhrase) {
+                findNavController().popBackStack()
+            } else {
+                showConfirmationDialog()
+            }
+        }
+
+        binding.saveButton.action = {
+            if (!isDefaultTextVisible()) {
+                binding.keyboardInput.text.let { text ->
+                    if (text.isNotBlank()) {
+                        viewModel.addNewPhrase(text.toString(), category.categoryId)
+                        savedPhrase = text.toString()
+                    }
+                }
+            }
+        }
+
+        (binding.phraseSavedView.root as TextView).apply {
+            setText(R.string.new_phrase_saved)
+        }
+
+        viewModel = ViewModelProviders.of(
+            requireActivity(),
```

Comment:
If this view model is only going to be used by this fragment, you can pass in `this` instead of `requireActivity`